thirdeye/thirdeye-frontend/app/pods/self-serve/create-alert/controller.js

```diff
@@ -366,15 +383,62 @@ export default Ember.Controller.extend({
       // The request failed. No graph to render.
       this.clearAll();
       this.setProperties({
-        isSelectMetricError: true,
-        isMetricDataInvalid: true,
         isMetricDataLoading: false,
         selectMetricErrMsg: error
       });
     });
   },
 
   /**
+   * If a dimension has been selected, the metric data object will contain subdimensions.
+   * This method calls for dimension ranking by metric, filters for the selected dimension,
+   * and returns a sorted list of graph-ready dimension objects.
+   * @method getTopDimensions
+   * @param {Object} data - the graphable metric data returned from fetchAnomalyGraphData()
+   * @param {Object} config - the graph configuration object
+   * @param {Number} maxSize - number of sub-dimensions to display on graph
+   * @param {String} selectedDimension - the user-selected dimension to graph
+   * @return {undefined}
+   */
+  getTopDimensions(data, config, maxSize, selectedDimension) {
+    const url = `/rootcause/query?framework=relatedDimensions&anomalyStart=${config.currentStart}&anomalyEnd=${config.currentEnd}&baselineStart=${config.baselineStart}&baselineEnd=${config.baselineEnd}&analysisStart=${config.currentStart}&analysisEnd=${config.currentEnd}&urns=thirdeye:metric:${config.id}&filters=${encodeURIComponent(config.filters)}`;
+    const colors = ['orange', 'teal', 'purple', 'red', 'green', 'pink'];
+    const dimensionObj = data.subDimensionContributionMap || {};
+    let dimensionList = [];
+    let topDimensions = [];
+    let topDimensionLabels = [];
+    let filteredDimensions = [];
+    let colorIndex = 0;
+
+    return new Ember.RSVP.Promise((resolve) => {
+      fetch(url).then(checkStatus)
+        .then((scoredDimensions) => {
+          // Select scored dimensions belonging the selected one
+          filteredDimensions =  _.filter(scoredDimensions, function(dimension) {
+            return dimension.label.split('=')[0] === selectedDimension;
+          });
+          // Prep a sorted list of labels for our dimension's top contributing sub-dimensions
+          topDimensions = filteredDimensions.sortBy('score').reverse().slice(0, maxSize);
+          topDimensionLabels = [...new Set(topDimensions.map(key => key.label.split('=')[1]))];
+          // Build the array of subdimension objects for the selected dimension
+          for(let subDimension of topDimensionLabels){
+            if (dimensionObj[subDimension] && subDimension !== '') {
```

Comment:
nit: empty strings are falsy, I feel `subDimension && dimensionObj[subDimension]` might be cleaner.

Reply:
Thanks.